DOCS-8778 Amazon SQS 5.8 Edits 2 (#998)
mulesoft/docs-connectors: amazon-sqs/5.8/modules/ROOT/pages/amazon-sqs-connector-reference.adoc

> [[config]]

Reviewer: BREAK here to send partial edit

Reviewer: RESUME

| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime purges the instances when as appropriate. |  | 
|===

=== Connection Types
[[config_basic]]
==== Basic

> 

Reviewer: Yikes, this is some heaving nesting of info.
I haven't looked at the context, but maybe we need to promote some of this to separate Reference topics.

Author: This is a generated referenced. It's just how it gets generated. Can't fix.

| Workstation a| String |  The optional proxy workstation. |  | 
|===
[[config_role]]
==== Role

====== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Role ARN a| String | Role ARN that uniquely identifies the role to assume to gain cross-account access |  | x

Reviewer: Period because of full sentences in same table column
```suggestion
| Role ARN a| String | Role ARN that uniquely identifies the role to assume to gain cross-account access. |  | x
```

> | Test Queue ARN a| String | Parameter used to test the connection to the SQS connector. If you do not specify this parameter the test connection is not performed. A queue ARN is constructed in the following format: arn:aws:{REGION}:{ACCOUNT_NUMBER}:{QUEUE_NAME}.  Example: arn:aws:sqs:us-east-1:123456789101:test-queue |  | 

Reviewer: ```suggestion
| Test Queue ARN a| String | Parameter used to test the connection to the SQS connector. If you do not specify this parameter, the test connection is not performed. A queue ARN is constructed in the following format: arn:aws:{REGION}:{ACCOUNT_NUMBER}:{QUEUE_NAME}, for example, arn:aws:sqs:us-east-1:123456789101:test-queue. |  | 
```

| Large Payload Support a| <<LargePayloadSupport>> |  |  | 
| Connection Timeout a| Number |  Amount of time to wait (in milliseconds) when initially establishing a connection before the connector times out. A value of 0 means infinity and is not recommended. |  50000 | 
| Max Connections a| Number |  Sets the maximum number of allowed open HTTP connections. |  | 
| Socket Timeout a| Number |  Amount of time to wait (in milliseconds) for data to be transferred over an established, open connection before the connection times out. A value of 0 means infinity and is not recommended. |  50000 | 
| Access Key a| String |  Access key provided by Amazon |  | x

> | Secret Key a| String |  Secret key provided by Amazon |  | x

Reviewer: ditto periods

| Try Default AWSCredentials Provider Chain a| Boolean |  Set this field to true to obtain credentials from the AWS environment, See: https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/credentials.html |  false | 

> | Region Endpoint a| String |  Sets the topic region endpoint |  US_EAST_1 | 

Reviewer: ditto

| TLS Configuration a| <<Tls>> |  |  | 
| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  | 
| Host a| String |  Optional proxy host |  | 
| Port a| Number |  Optional proxy port |  | 
| Username a| String |  Optional proxy username |  | 
| Password a| String |  Optional proxy password |  | 
| Domain a| String |  Optional proxy domain |  | 

> | Workstation a| String |  Optional proxy workstation |  | 

Reviewer: ditto

=== Add Permission
`<sqs:add-permission>`

This operation adds a permission to a message queue.

==== Parameters
[%header,cols="20s,20a,35a,20a,5a"]
|===
| Name | Type | Description | Default Value | Required

> | Configuration | String | Name of the configuration to use | | x

Reviewer: ditto

| Label a| String |  Name for this permission |  | x
| Account Ids a| Array of String |  IDs of the AWS accounts to share this queue with |  | x
| Actions a| Array of String |  List to indicate how much to share (SendMessage, ReceiveMessage, ChangeMessageVisibility, DeleteMessage, GetQueueAttributes) |  | x

> | Queue Url a| String |  Permissions will be added to the queue represented by this URL. This parameter is optional, and if you do not specify `queueUrl`, you must specify the Default Global Queue URL at the configuration level. |  | 

Reviewer: ```suggestion
| Queue Url a| String |  Permissions are added to the queue represented by this URL. This parameter is optional, and if you do not specify `queueUrl`, you must specify the default global queue URL at the configuration level. |  | 
```

> | Target Variable a| String | Name of a variable in which to store the operation's output |  | 

Reviewer: ditto

Reviewer: I'm going to stop marking these; please give all items in one column a period if even one of them ends with on.
Same with items in a list.

** MINUTES
** HOURS
** DAYS |  |  SECONDS | 

> | Queue Url a| String |  URL of the Amazon SQS queue to act on. This parameter is optional, and if you do not specify `queueUrl`, you must specify the Default Global Queue URL at the configuration level. |  | 

Reviewer: I would globally lowercase "the default global queue URL."

|===
| Field | Type | Description | Default Value | Required
| Bucket a| String | Name of the AWS S3 bucket in which to store large payload messages.
 The AWS S3 bucket must already be created and configured in AWS S3. Enabling this feature incurs additional charges for using AWS S3. |  | x
| Message Size Threshold a| Number | The message size threshold value for storing message payloads in the AWS S3 bucket.

>  The default value for the message size threshold is 256 KB and the maximum threshold size value is 256KB. The maximum message size is 2GB. | 256 | 

Reviewer: ```suggestion
 The default value for the message size threshold is 256 KB and the maximum threshold size value is 256 KB. The maximum message size is 2 GB. | 256 | 
```
